splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java

```diff
@@ -2145,4 +2146,349 @@ public static void INVALIDATE_GLOBAL_DICTIONARY_CACHE() throws Exception {
             }
         }
     }
+
+    public static void SHOW_CREATE_TABLE(String schemaName, String tableName, ResultSet[] resultSet) throws SQLException
+    {
+        try {
+            Connection connection = getDefaultConn();
+            Statement stmt = connection.createStatement();
+
+            EngineUtils.verifyTableExists(connection, schemaName, tableName);
+
+            ResultSet tableIdRs = stmt.executeQuery("SELECT T.TABLEID, T.TABLETYPE, T.COMPRESSION, T.DELIMITED, " +
+                    "T.ESCAPED, T.LINES, T.STORED, T.LOCATION" +
+                    " FROM SYS.SYSTABLES T, SYS.SYSSCHEMAS S " +
+                    "WHERE T.TABLETYPE IN ('T','E') AND T.SCHEMAID = S.SCHEMAID " +
+                    "AND T.TABLENAME = '" + tableName + "' AND S.SCHEMANAME = '" + schemaName + "'");
+
+            PreparedStatement getColumnInfoStmt = connection.prepareStatement("SELECT C.COLUMNNAME, C.REFERENCEID, " +
+                    "C.COLUMNNUMBER FROM SYS.SYSCOLUMNS C, SYS.SYSTABLES T WHERE T.TABLEID = ? " +
+                    "AND T.TABLEID = C.REFERENCEID ORDER BY C.COLUMNNUMBER");
+
+            PreparedStatement getPartitionedColsStmt = connection.prepareStatement("SELECT C.COLUMNNAME " +
+                    "FROM SYS.SYSCOLUMNS C, SYS.SYSTABLES T WHERE T.TABLEID = ? " +
+                    "AND T.TABLEID = C.REFERENCEID AND C.PARTITIONPOSITION > -1 ORDER BY C.PARTITIONPOSITION");
+
+            String tableId = "" ;
+            boolean firstCol = true;
+
+            //External Table
+            String isExternal = "";
+            StringBuilder extTblString = new StringBuilder("");
+            while (tableIdRs.next()){
```

Comment:
The while loop suggests that we could be handling more than one table here. If we have more than one external tables, the specifics for the multiple external tables will be lumped together. Or maybe what you want is an IF instead of WHILE here?

Reply:
Also intending to support multiple tables.
Change the WHILE to IF  and reorganize scope.